components/CardList/index.js

```diff
@@ -5,13 +5,22 @@ import { useContext } from "react";
 import LoadingSpinner from "../LoadingSpinner";
 import FilterSection from "../FilterButtonSection";
 import { filterButtonsData } from "../../utils/data/filterButtonsData";
+import { useState } from "react";
+import { useEffect } from "react";
+import SearchField from "../SearchField";
 
 export default function CardList() {
   const breedData = useContext(BreedData);
-  const activeButtonId = useContext(Filter);
+  const { activeButtonId, setActiveButtonId } = useContext(Filter);
+
+  const [searchTerm, setSearchTerm] = useState("");
+
+  useEffect(() => {
+    setActiveButtonId({ id: 1 });
+  }, [searchTerm]);
```

Comment:
You are setting a state as the result of another state changing. This is generally an antipattern. You currently have no other way of doing this because you're updating something in context, but you should consider restructuring this more generally.

See: https://react.dev/learn/you-might-not-need-an-effect#chains-of-computations